MaxText/inference/paged_attention.py

```diff
@@ -125,26 +127,68 @@ def paged_dot_product_attention_with_max_and_sum(self, query, key, value):
 
     return attn, local_max, local_sums
 
-  def paged_attention(
+  # TODO(rupliu): add sharding when SPMD is fully supported
+  def paged_attention_v2_prefill(
       self,
       query: Array,
       key_pages_var: nn.Variable,
       value_pages_var: nn.Variable,
       page_state: page_manager.PageState,
   ) -> Array:
-    """Apply Paged Attention.
-
-    Annotations:
-    b: batch_size
-    s: sequence length
-    n: query heads
+    """Apply ragged input Paged Attention in prefill only."""
+    k_p = jnp.permute_dims(key_pages_var.value, (1, 2, 0, 3))
```

Comment:
nit: Mind adding a comment for the expected shape of key_pages_var

num_kv_heads, num_pages, tokens_per_page, kv_head_dim_size

Reply:
updated!